src/io/flutter/run/daemon/DeviceDaemon.java

```diff
@@ -315,6 +362,15 @@ public void onDeviceRemoved(@NotNull DaemonEvent.DeviceRemoved event) {
       deviceChanged.run();
     }
 
+    @Override
+    public void processTerminated(int exitCode) {
+      if (running.get()) {
+        processStopped.accept(
+          "Daemon #" + daemonId + " exited. Exit code: " + exitCode + ". Stderr:\n" +
+          api.getStderrTail());
```

Comment:
I don't think it's necessary; it will be garbage collected when we remove the last reference to the DeviceDaemon. 

A project should only reference one DeviceDaemon at a time, except transiently when the DaemonService is attempting to start a new one.